mcs/class/System/Test/System.Net/HttpWebRequestTest.cs

```diff
@@ -2301,6 +2303,70 @@ public void BeginGetResponseAborts ()
 
 			return;
 		}
+		
+		[Test]
+		public void TestLargeDataReading ()
+		{
+			int near2GBStartPosition = rand.Next (int.MaxValue - 500, int.MaxValue);
+			AutoResetEvent readyGetLastPortionEvent = new AutoResetEvent (false);
+			AssertionException testException = null;
+
+			DoRequest (
+			(request, waitHandle) =>
+			{
+				try
+				{
+					const int timeoutMs = 5000;
+
+					request.Timeout = timeoutMs;
+					request.ReadWriteTimeout = timeoutMs;
+
+					WebResponse webResponse = request.GetResponse ();
+					Stream webResponseStream = webResponse.GetResponseStream ();
+					Assert.IsNotNull (webResponseStream, null, "#1");
+					
+					Type webConnectionStreamType = webResponseStream.GetType ();
+					FieldInfo totalReadField = webConnectionStreamType.GetField ("totalRead", BindingFlags.NonPublic | BindingFlags.Instance);
+					Assert.IsNotNull (totalReadField, "#2");
+					totalReadField.SetValue (webResponseStream, near2GBStartPosition);
+					
+					byte[] readBuffer = new byte[int.MaxValue - near2GBStartPosition];
+					Assert.AreEqual (webResponseStream.Read (readBuffer, 0, readBuffer.Length), readBuffer.Length, "#3");
+					readyGetLastPortionEvent.Set ();
+					Assert.Greater (webResponseStream.Read (readBuffer, 0, readBuffer.Length), 0, "#4");
+					readyGetLastPortionEvent.Set ();
+					
+					webResponse.Close();
+				}
+				catch (AssertionException e)
```

Comment:
Please change this to catch `Exception`, as otherwise we would ignore any error that doesn't come from an Assert.